src/v0/sources/shopify V2/commonUtils.js

```diff
@@ -0,0 +1,129 @@
+const { RedisDB } = require('../../../util/redis/redisConnector');
+const logger = require('../../../logger');
+const {
+  lineItemsMappingJSON,
+  productMappingJSON,
+  LINE_ITEM_EXCLUSION_FIELDS,
+  PRODUCT_MAPPING_EXCLUSION_FIELDS,
+  SHOPIFY_TRACK_MAP
+} = require('./config');
+const { TransformationError } = require('../../util/errorTypes');
+
+const getCartToken = message => {
+  const { event } = message;
+  if (event === SHOPIFY_TRACK_MAP.carts_update) {
+    return message.properties?.id || message.properties?.token;
+  }
+  return message.properties?.cart_token || null;
+};
+
+const getDataFromRedis = async (key, metricMetadata) => {
+  try {
+    stats.increment('shopify_redis_calls', {
+      type: 'get',
+      field: 'all',
+      ...metricMetadata,
+    });
+    const redisData = await RedisDB.getVal(key);
+    if (redisData === null) {
+      stats.increment('shopify_redis_no_val', {
+        ...metricMetadata,
+      });
+    }
+    return redisData;
+  }
+  catch (e) {
+    logger.debug(`{{SHOPIFY::}} Get call Failed due redis error ${e}`);
+    stats.increment('shopify_redis_failures', {
+      type: 'get',
+      ...metricMetadata,
+    });
+  }
+  return null;
+};
+
+/**
+ * query_parameters : { topic: ['<shopify_topic>'], ...}
+ * Throws error otherwise
+ * @param {*} event
+ * @returns
+ */
+const getShopifyTopic = (event) => {
+  const { query_parameters: qParams } = event;
+  logger.debug(`[Shopify] Input event: query_params: ${JSON.stringify(qParams)}`);
+  if (!qParams) {
+    throw new TransformationError('Query_parameters is missing');
+  }
+  const { topic } = qParams;
+  if (!topic || !Array.isArray(topic)) {
+    throw new TransformationError('Invalid topic in query_parameters');
+  }
+  if (topic.length === 0) {
+    throw new TransformationError('Topic not found');
+  }
+  return topic[0];
+};
+
+const getHashLineItems = (cart) => {
+  if (cart && cart?.line_items && cart.line_items.length > 0) {
+    return sha256(JSON.stringify(cart.line_items));
+  }
+  return 'EMPTY';
+};
+
+const getVariantString = (lineItem) => {
+  const { variant_id, variant_price, variant_title } = lineItem;
+  return `${variant_id || ''} ${variant_price || ''} ${variant_title || ''}`;
+};
+
+const getProductsListFromLineItems = (lineItems) => {
+  if (!lineItems || lineItems.length === 0) {
+    return [];
+  }
+  const products = [];
+  lineItems.forEach((lineItem) => {
+    const product = constructPayload(lineItem, lineItemsMappingJSON);
+    extractCustomFields(lineItem, product, 'root', LINE_ITEM_EXCLUSION_FIELDS);
+    product.variant = getVariantString(lineItem);
+    products.push(product);
+  });
+  return products;
+};
+
+const createPropertiesForEcomEvent = (message) => {
+  const { line_items: lineItems } = message;
+  const productsList = getProductsListFromLineItems(lineItems);
+  const mappedPayload = constructPayload(message, productMappingJSON);
```

Comment:
We need a separate mapping for each event. Can't use productMappingJSON for all of the ecomm events. We need to add the support of using different mapping files for different events.

Reply:
Agreed.